letsencrypt-auto

```diff
@@ -35,14 +35,11 @@ done
 # letsencrypt itself needs root access for almost all modes of operation
 # The "normal" case is that sudo is used for the steps that need root, but
 # this script *can* be run as root (not recommended), or fall back to using
-# `su`
-if test "`id -u`" -ne "0" ; then
-  if command -v sudo 1>/dev/null 2>&1; then
-    SUDO=sudo
-  else
-    echo \"sudo\" is not available, will use \"su\" for installation steps...
+# `su`. Auto-detection can be overrided by explicitly setting the
+# environment variable SUDO to 'sudo', 'sudo_su' or '' as used below.
+init_su_sudo() {
```

Comment:
No, I don't see a downside. I think the main reason I wrote it like that was that it required less code line changes.